### ISSUE TITLE
Prediction part code

### ISSUE THREAD

**Issue by lighnt** [davidBelanger/SPEN#1]
Hi, could you please tell me where is the code for solving equation (2) of your paper? I spent a long time studying your code but failed to find it.

**Comment by davidBelanger** [comment]
eq (2) specifies an optimization problem, but does not describe an algorithm for solving it. The algorithm we use is gradient descent. We instantiate this algorithm as a long neural network that implements gradient descent for a fixed number of iterations. The advantage of implementing the algorithm as a differentiable neural network is that we can perform end-to-end training. 

This network, which is api-equivalent to a simple feed-forward classifier, is constructed in main.lua as problem.inferencer.

**Comment by lighnt** [comment]
Thanks a lot, :)
